javaparser-core/src/main/java/com/github/javaparser/printer/lexicalpreservation/TokenTextElement.java

```diff
@@ -81,27 +65,25 @@ public boolean equals(Object o) {
 
         TokenTextElement that = (TokenTextElement) o;
 
-        if (tokenKind != that.tokenKind) return false;
-        return text.equals(that.text);
+        if (token.getKind() != that.token.getKind()) return false;
+        return getText().equals(that.getText());
 
     }
 
     @Override
     public int hashCode() {
-        int result = tokenKind;
-        result = 31 * result + text.hashCode();
-        return result;
+        return token.hashCode();
     }
 
     @Override
     public String toString() {
-        return "TokenTextElement(" + tokenKind +
-                ") {" + text + '}';
+        return "TokenTextElement(" + token.getKind() +
+                ") {" + getText() + '}';
```

Comment:
@ftomassetti I could use JavaToken's toString here - are you very attached to this representation?